lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -141,6 +141,143 @@ LogicalResult ReturnOp::verify() {
 
   return success();
 }
+
+//===----------------------------------------------------------------------===//
+// InstanceOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult InstanceOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto targetClass = getClass(&symbolTable);
+  if (!targetClass)
+    return emitOpError() << "'" << getClassName() << "' does not exist";
+
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// PortReadOp
+//===----------------------------------------------------------------------===//
+
+template <typename TSrcOp, typename TTargetOp>
+LogicalResult verifyPortSymbolUses(
+    TSrcOp op, StringAttr symName, llvm::function_ref<Type(TSrcOp)> getPortType,
+    SymbolTableCollection &symbolTable,
+    llvm::function_ref<FailureOr<TTargetOp>(Operation *)> getTargetOp) {
+
+  ClassOp parentClass = op->template getParentOfType<ClassOp>();
+  if (!parentClass)
+    return op->emitOpError()
+           << op->getName() << " must be contained in a ClassOp";
+
+  Operation *rootAccessOp = SymbolTable::lookupSymbolIn(parentClass, symName);
```

Comment:
This is a linear scan through the parentClass, right? If so, this is unacceptably inefficient longer term. Once we get the innerSym table stuff (on which I am pushing), it'll be more efficient. Please add a TODO comment reflecting this.

Also, doesn't the SymbolTable interface expose a method to do this?